### PR TITLE
[Draft]Add doc for category and tags.

### PR DESCRIPTION
**Pull request by jiazengcindy**
# Description

Add doc for category and tags.

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Add Tool Category and Tags
+Users sometimes need their tools to be easy to find. To achieve this, we've introduced the `category` and `tags`. The tool category helps organize tools into specific category folders, while tool tags can be used to search for tools with similar tags, even if they belong to different categories.   
+Both category and tags are optional. If a tool doesn't have a category, it will be displayed in the root folder. If there are no tags, the tags field will be empty.
+
+## Prerequisites
+- Please ensure that your [Prompt flow for VS Code](https://marketplace.visualstudio.com/items?itemName=prompt-flow.prompt-flow) is updated to version 1.1.0 or later.
+
+## How to create a tool with category and tags
+Here we use [an existing tool package](https://github.com/microsoft/promptflow/tree/main/examples/tools/tool-package-quickstart/my_tool_package) as an example. If you want to create your owner tool, please refer to [create and use tool package](create-and-use-tool-package.md#create-custom-tool-package), and you can add the _category_ and _tags_ fields in the tool's YAML, the YAML should like following:
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md:9]: [an existing tool](refer to the tool yaml directly)

Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md:9]: your own tool (not owner tool)

File: docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      type:
+      - string
+  module: my_tool_package.tools.tool_with_file_path_input
+  name: Tool with FilePath Input
+  description: This is a tool to demonstrate the usage of FilePath input
+  type: python
+  category: test_tool
+  tags:
+    input: FilePath
+    task: text-generation
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md:27]: Add a comment before line24:
# add category and tags like below

File: docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md (hunk 1 above)
Comment (jiazengcindy) [reply docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md]: Changed, thanks.

Comment (jiazengcindy) [reply docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md]: Changed, thanks.

File: docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md (hunk 2 above)
Comment (jiazengcindy) [reply docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md:28]: Added, thanks.

File: docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Add Tool Category and Tags
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md:1]: Add Category and Tags for Tool ?

File: docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+Users sometimes need their tools to be easy to find. To achieve this, we've introduced the `category` and `tags`. The tool category helps organize tools into specific category folders, while tool tags can be used to search for tools with similar tags, even if they belong to different categories.   
+Both category and tags are optional. If a tool doesn't have a category, it will be displayed in the root folder. If there are no tags, the tags field will be empty.
+
+## Prerequisites
+- Please ensure that your [Prompt flow for VS Code](https://marketplace.visualstudio.com/items?itemName=prompt-flow.prompt-flow) is updated to version 1.1.0 or a more recent version.
+
+## How to create a tool with category and tags
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md:8]: How to add category and tags for a tool ?

File: docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  description: This is a tool to demonstrate the usage of FilePath input
+  type: python
+  # add category and tags like below
+  category: test_tool
+  tags:
+    input: FilePath
+    task: text-generation
+```
+
+## Use tool category and tags
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md:31]: Tool with category and tags shown in VS Code extension

File: docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+### Tool category and tags in VS Code extension
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md:32]: Tool with category and tags shown in tool tree

File: docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+Follow [steps](create-and-use-tool-package.md#use-your-tool-from-vscode-extension) to use your tool from VS Code extension. Your tool will display with category and tags:  
+![category_and_tags_in_extension](../../media/how-to-guides/develop-a-tool/category_and_tags_in_extension.png)
+
+### Tool category and tags in tool list
```
Comment (dorisjoy) [inline docs/how-to-guides/develop-a-tool/add-tool-category-and-tags.md:36]: Tool with category and tags shown in tool list

**Pull request closed without merging**